### ISSUE TITLE
Remove Unmaintained & Broken Features

### ISSUE THREAD

**Issue by michael-schwarz** [goblint/cil#28]
During his thesis @keremc identified various features of CIL (mostly in `ext`) that either:

- are only useful in conjunction with some other, now defunct projects
- display unsound behavior in a variety of cases (e.g. `partial` which is unsound in the presence of pointers)

**In the interest of keeping the code base to maintain small, I suggest we remove these features before we cut the next release of `goblint-cil`.** As a next step `dopartial` would then also need to be removed from Goblint.

This does not make `goblint-cil` any more incompatible with original `cil` than it already is, apart from the fact that some extra features of `cil` will not be available in `goblint-cil`

@keremc can you attach the PDF of your thesis to this issue for reference?

**Comment by raveenkm** [comment]
Hi, for my analysis I found Simplify as useful transformation. Can I take the modules from original CIL and build it using goblint-cil ?

**Comment by michael-schwarz** [comment]
Hi, I think your best bet is taking the version of `simplify` we had just before we removed it https://github.com/goblint/cil/pull/30/files, and then try to see what additional changes are needed for it to compile.

It should mostly be fairly mechanical changes, except maybe in cases where we have added support for new C11 constructs by adding new variants for some types.

If you're stuck somewhere, let us know, we might be able to lend a hand!
